snippets/spring-data/query.json

```diff
@@ -0,0 +1,32 @@
+        "description": "Use SQL++ to Query the Couchbase Cluster, this has all options available."
+    },
+    "SQL++ Query With Parameters": {
+        "prefix": ["@cbsdqparam"],
+        "body": ["public static N1QLQuery buildQueryParams(N1QLExpression expression, JsonValue queryPlaceholderValues,","QueryScanConsistency scanConsistency) {","    QueryOptions opts = QueryOptions.queryOptions().scanConsistency(scanConsistency);","","    // put the placeholders in the options","    if (queryPlaceholderValues instanceof JsonObject && !((JsonObject) queryPlaceholderValues).isEmpty()) {","        opts.parameters((JsonObject) queryPlaceholderValues);","    } else if (queryPlaceholderValues instanceof JsonArray && !((JsonArray) queryPlaceholderValues).isEmpty()) {","        opts.parameters((JsonArray) queryPlaceholderValues);","    }","","    return new N1QLQuery(expression, opts);","}","","N1QLQuery query = buildQueryParams(${1:expression}, ${2:queryPlaceholderValues}, ${3:scanConsistency});","${4:CouchbaseTemplate}.findByQuery(${5:Model}.class).matching(query);"],
+        "description": "Use parameterised SQL++ to query the Couchbase Cluster."
+    },
+    "SQL++ Query with FTS": {
+        "prefix": ["@cbsdqfts"],
+        "body": [],
+        "description": "Use SQL++ to query the Couchbase Cluster with full text search."
+    },
+    "SQL++ Query Import": {
+        "prefix": ["@cbsdqimp"],
+        "body": ["import java.time.Duration;","","import org.springframework.beans.factory.annotation.Autowired;","import org.springframework.data.couchbase.core.CouchbaseTemplate;","import org.springframework.data.couchbase.core.query.N1QLExpression;","import org.springframework.data.couchbase.core.query.N1QLQuery;","import org.springframework.data.couchbase.core.query.Query;","import org.springframework.data.couchbase.core.query.QueryCriteria;","","import com.couchbase.client.core.deps.com.fasterxml.jackson.annotation.JsonValue;","import com.couchbase.client.java.codec.JsonSerializer;","import com.couchbase.client.java.json.JsonArray;","import com.couchbase.client.java.json.JsonObject;","import com.couchbase.client.java.kv.MutationState;","import com.couchbase.client.java.query.QueryOptions;","import com.couchbase.client.java.query.QueryProfile;","import com.couchbase.client.java.query.QueryScanConsistency;","import com.couchbase.client.java.search.SearchScanConsistency;"],
+        "description": "Import all necessary query namespaces"
```

Comment:
"packages" for Java.
And would be good if the descriptions were consistent.  Minor thing but most of them end in a "." and this doesn't.